SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -289,19 +317,15 @@ TOptional<Trace_SpanId> SpatialEventTracer::PopLatentPropertyUpdateSpanId(const
 		return {};
 	}
 
-	FSpatialSpanIdStack* Stack = ObjectSpanIdStacks.Find(Object);
-	if (Stack == nullptr)
+	Trace_SpanId* ExistingSpanId = ObjectSpanIdStacks.Find(Object);
+	if (ExistingSpanId == nullptr)
 	{
 		return {};
 	}
 
-	TOptional<Trace_SpanId> SpanId = Stack->Pop();
-	if (Stack->IsEmpty())
-	{
-		ObjectSpanIdStacks.Remove(Object);
-	}
+	ObjectSpanIdStacks.Remove(Object);
 
-	return SpanId;
+	return *ExistingSpanId;
```

Comment:
This pointer gets invalidated by `ObjectSpanIdStacks.Remove`. You need to deference and save into a local variable before calling `Remove`.